third_party/mbedtls/CMakeLists.txt

```diff
@@ -26,15 +26,37 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-set(OT_MBEDTLS_DEFAULT_CONFIG_FILE \"${CMAKE_CURRENT_SOURCE_DIR}/mbedtls-config.h\")
+set(OT_MBEDTLS_DEFAULT_CONFIG_FILE \"${CMAKE_CURRENT_BINARY_DIR}/mbedtls-config.h\")
 
 set(OT_MBEDTLS_CONFIG_FILE "" CACHE STRING "The mbedTLS config file")
 
 set(ENABLE_TESTING OFF CACHE BOOL "Disable mbedtls test" FORCE)
 set(ENABLE_PROGRAMS OFF CACHE BOOL "Disable mbetls program" FORCE)
 
+find_program(UNIFDEFALL_EXE unifdefall)
+find_program(SED_EXE sed)
+
 add_subdirectory(repo)
 
+if(UNIFDEFALL_EXE AND SED_EXE)
+    add_custom_target(mbedtls-config.h
+        ${UNIFDEFALL_EXE}
+            "'-D$<JOIN:$<TARGET_PROPERTY:ot-config,INTERFACE_COMPILE_DEFINITIONS>,';'-D>'"
```

Comment:
nit: are single quotes necessary here? I noticed that the lines below do not use single quotes.

Reply:
Yes, they are necessary here. This makes definitions like `-DOPENTHREAD_PROJECT_CORE_CONFIG_FILE="openthread-core-simulation-config.h"` passed correctly. Without this the double quotes will be interpreted by shell and eventually the macro will be expanded to a string without quotes.